actors/miner/src/lib.rs

```diff
@@ -4857,10 +4868,12 @@ fn confirm_sector_proofs_valid_internal(
             e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to put new sectors")
         })?;
 
+        // XXX will need to filter by new sectors when extending to handle PRU
         state.delete_precommitted_sectors(store, &new_sector_numbers).map_err(|e| {
             e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to delete precommited sectors")
         })?;
 
+        // XXX will need to filter by new sectors when extending to handle PRU
```

Comment:
As an alternative, perhaps this ActivateSectors should be broken into two halves, where PRU only calls the first half, and PCA calls both; the second half deals with precommits and assignment. Activating a sector and activating some data/power/pledge in a sector are two different things. So rather than a flag to switch off half the method, two methods for callers to compose.
The first half might even have a parameter like DataActivation, excised from SectorActivation and extended with an expiration epoch, etc.